tools/ipapy2/ipapy2_MS1_annotation.xml

```diff
@@ -0,0 +1,105 @@
+    
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">ipapy2</requirement>
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+        python3 '${ipapy2_MS1_annotation_cli}'
+    ]]></command>
+
+<configfiles>
+<configfile name="ipapy2_MS1_annotation_cli">
+@init_logger@
+
+df = pd.read_csv('${MS1_table}')
+allAdds = pd.read_csv('${all_adducts}' )
+#if $ppmunk == "None"
+#if $ppmthr == "None"
+annotations = ipa.MS1annotation(df, allAdds, ppm=$ppm, me=$me, ratiosd=$ratiosd, ppmunk=$ppm, ratiounk=$ratiounk, ppmthr=2*$ppm, pRTNone=$pRTNone, pRTout=$pRTout, ncores=$ncores)
+#else
+annotations = ipa.MS1annotation(df, allAdds, ppm=$ppm, me=$me, ratiosd=$ratiosd, ppmunk=$ppm, ratiounk=$ratiounk, ppmthr=$ppmthr, pRTNone=$pRTNone, pRTout=$pRTout, ncores=$ncores)
+#end if
+#else
+#if $ppmthr == "None"
+annotations = ipa.MS1annotation(df, allAdds, ppm=$ppm, me=$me, ratiosd=$ratiosd, ppmunk=$ppmunk, ratiounk=$ratiounk, ppmthr=2*$ppm, pRTNone=$pRTNone, pRTout=$pRTout, ncores=$ncores)
+#else
+annotations = ipa.MS1annotation(df, allAdds, ppm=$ppm, me=$me, ratiosd=$ratiosd, ppmunk=$ppmunk, ratiounk=$ratiounk, ppmthr=$ppmthr, pRTNone=$pRTNone, pRTout=$pRTout, ncores=$ncores)
+#end if
+#end if
+annotations.to_csv('${output}', index=False)
+</configfile>
+</configfiles>
+
+    <inputs>
+        <param label="MS1 table" name="MS1_table" type="data" format="csv" help="pandas dataframe containing the MS1 data." />
+        <param label="all possible adducts table" name="all_adducts" type="data" format="csv" help="pandas dataframe containing the information on all the possible adducts given the database." />
+        <param label="ppm" name="ppm" type="float" help="accuracy of the MS instrument used."/>
+        <param label="me" name="me" type="float" value="0.000548579909065" help="accurate mass of the electron. Default 5.48579909065e-04."/>
+        <param label="ratiosd" name="ratiosd" type="float" value="0.9" help="It represents the acceptable ratio between predicted intensity and observed intensity of isotopes. default 0.9."/>
+        <param label="ppmunk" name="ppmunk" type="float" optional="true" value="None" help="ppm associated to the 'unknown' annotation. If not provided equal to ppm."/>
+        <param label="ratiounk" name="ratiounk" type="float" value="0.5" help="isotope ratio associated to the 'unknown' annotation. If not provided equal to 0.5."/>
+        <param label="ppmthr" name="ppmthr" type="float" optional="true" value="None" help="Maximum ppm possible for the annotations. Ff not provided equal to 2*ppm."/>
+        <param label="pRTNone" name="pRTNone" type="float" value="0.8" help="Multiplicative factor for the RT if no RTrange present in the database. If not provided equal to 0.8."/>
+        <param label="pRTout" name="pRTout" type="float" value="0.4" help="Multiplicative factor for the RT if measured RT is outside the RTrange present in the database. If not provided equal to 0.4."/>
```

Comment:
Are these some RT transformation, or how is this supposed to be used? Is it like a linear scaling of RT?